src/components/Theme/src/Theme.vue

```diff
@@ -20,21 +21,66 @@ function isVueComponent(value) {
 	return value && value.render;
 }
 
+// returns true if value is an array
+// of objects with id fields
+function isKeyedArray(value) {
+	return value?.[0]?.id;
+}
+
+function keyedArrayIntoObject(arrayValue) {
+	const objectValue = {};
+	for (const item of arrayValue) {
+		objectValue[item.id] = item;
+	}
+	return objectValue;
+}
+
+function objectIntoKeyedArray(objectValue) {
```

Comment:
There was an open question regarding why we chose to define profiles in an array. I can't recall that reason and we likely need to continue to support that to avoid breaking changes, but I could imagine all this could get much simpler without having to iterate through each.

Should we document how / why we support this data format? Should we consider changing it?

Reply:
I am guessing it was set up as an array to match Website. Had to dig pretty deep to find the reason profiles are saved as an array in Website instead of an object:
https://coda.io/d/ECOM-Website_dM7hmP2aSPk/Color-schema-updates_su4Up#_luUPn

```
Previous colorProfiles structure: Object instead of array
Faster, easier for lookup, but if we allow customer driven creation of new custom profiles in the future, we’ll need a way to track their order so they display consistently when displayed as options in the UI, and the object approach displayed below would need to be iterated on.
```
<img width="829" alt="Screen Shot 2022-06-15 at 10 16 04 AM" src="https://user-images.githubusercontent.com/1486885/173849661-6a72b225-2ced-4d41-b19c-69df56964946.png">

Obviously, this is not a consideration for Maker themes at all so I think we're fine to convert `theme.profiles` to an object down the line.

Reply:
i think this is the key reason, and really the only reason, why an array was chosen over an object:

> we’ll need a way to **track their order** so they **display consistently when displayed as options in the UI**

that's the only thing an array can do that an object can't, unless we generate ascii-betical ids, e.g. `--profile-01`, `--profile-02` and then sort by their ids to generate a consistent ordering.

Reply:
anyway, this discussion is outside the scope of this PR, since if we do migrate profiles from a keyedArray into an object that would be a breaking change and force refactors in website (and possible checkout) which are both fine things to do but something we can do in a follow-up PR